pkg/build/builder/daemonless.go

```diff
@@ -45,6 +55,70 @@ func dropCapabilities() []string {
 	return dropCapabilities
 }
 
+// parsePullCredentials parses credentials from provided file.
+func parsePullCredentials(credsPath string) (credentialprovider.DockerConfig, error) {
+	var creds credentialprovider.DockerConfig
+	var err error
+
+	if filepath.Base(credsPath) == dockercfg.DockerConfigKey {
+		if creds, err = credentialprovider.ReadDockercfgFile(
+			[]string{filepath.Dir(credsPath)},
+		); err != nil {
+			return nil, err
+		}
+	} else {
+		if creds, err = credentialprovider.ReadSpecificDockerConfigJsonFile(
+			credsPath,
+		); err != nil {
+			return nil, err
+		}
+	}
+
+	if creds == nil {
+		creds = make(map[string]credentialprovider.DockerConfigEntry)
+	}
+
+	return creds, nil
+}
+
+// mergeNodeCredentials merges node credentials with credentials file provided.
+//
+// Node pull credentials are merged with credentials present on file path
+// provided as parameter, this function returns a new file path where merged
+// credentials are stored.
+func mergeNodeCredentials(credsPath string) (string, error) {
+	nodeCreds, err := parsePullCredentials(nodeCredentialsFile)
+	if err != nil {
+		log.V(2).Infof("proceeding without node credentials: %v", err)
+	}
+
+	namespaceCreds, err := parsePullCredentials(credsPath)
+	if err != nil {
+		return "", fmt.Errorf("error reading pull credentials: %v", err)
+	}
+
+	for regurl, cfg := range nodeCreds {
```

Comment:
for regurl, cfg := range nodeCreds {
		if _, ok := namespaceCreds[regurl]; !ok {
			namespaceCreds[regurl] = cfg
		}
	}

Reply:
Right, fixed.